numba/cuda/cudaimpl.py

```diff
@@ -641,7 +641,12 @@ def _generic_array(context, builder, shape, dtype, symbol_name, addrspace,
         # Create global variable in the requested address-space
         gvmem = lmod.add_global_variable(laryty, symbol_name, addrspace)
         # Specify alignment to avoid misalignment bug
-        gvmem.align = context.get_abi_sizeof(lldtype)
+        align = context.get_abi_sizeof(lldtype)
+        # Alignment is required to be a power of 2 for shared memory. If it is
+        # not a power of 2 (e.g. for a Record array) then round up accordingly.
+        if (align & (align - 1)) != 0:
+            align = 2 ** math.ceil(math.log2(align))
```

Comment:
Is this `align = 1 << (align - 1 ).bit_length())` ?

Reply:
I believe so.